# Pandas Applied — 주문 데이터 분석

## 업무 요청

온라인 스토어의 한 달치 주문 데이터를 받았습니다.

> **이번 달 무엇이 매출을 만들었고, 이 데이터를 어디까지 믿어도 될까요?**

코드 실행에서 끝나지 않고 이 질문에 답할 Evidence를 만듭니다.

In [ ]:
# colab 사용하시는 분들만!!!!
# 아래 코드 주석 해제해서 돌리셈

# import os
# os.makedirs('data', exist_ok=True)
#
# from google.colab import files
# uploaded = files.upload()   # 파일 선택창이 뜹니다
#
# # 업로드된 파일을 data/ 로 옮기기
# for name in uploaded:
#     os.rename(name, f'data/{name}')
#
# print(os.listdir('data'))

In [2]:
import pandas as pd

# Mission 1 — 데이터 읽기

`data/orders.csv`를 읽어 `orders`에 저장해보세요.

In [21]:
# CSV를 읽어 orders 변수에 저장하세요.
pass

orders = pd.read_csv("data/orders.csv")

<details>
<summary>힌트 보기</summary>

`pd.read_csv("data/orders.csv")` 형태를 떠올려보세요.

</details>

In [ ]:
orders = pd.read_csv("data/orders.csv")

## 바로 분석해도 될까요?

먼저 데이터의 처음 몇 행, 크기, 열, 결측 여부와 dtype을 확인하세요.

In [22]:
# 처음 몇 행을 확인하세요.
pass

print("orders.shape:", orders.shape)
cols = list(orders.columns)
print("orders.columns :", cols)
print("orders.dtypes: ", orders.dtypes)
print(orders.head(5))

orders.shape: (1000, 13)
orders.columns : ['order_id', 'customer_id', 'order_date', 'product_id', 'product_name', 'category', 'price', 'quantity', 'discount', 'payment_method', 'region', 'channel', 'record_status']
orders.dtypes:  order_id              str
customer_id           str
order_date            str
product_id            str
product_name          str
category              str
price               int64
quantity            int64
discount          float64
payment_method        str
region                str
channel               str
record_status         str
dtype: object
    order_id customer_id  order_date product_id product_name category  price  \
0  ORD000001       C0291  2026-07-16       A004          토트백       패션  32000   
1  ORD000002       C0254  2026-07-31       A003            캡       패션  25000   
2  ORD000003       C0223  2026-07-05       S004          플래너       문구  12000   
3  ORD000004       C0074  2026-07-12       E002          마우스     전자기기  29000   
4  ORD000005     

In [ ]:
# 몇 행 몇 열인지 확인하세요.
pass

In [ ]:
# 열과 결측 여부, dtype을 한 번에 확인해보세요.
pass

In [23]:
orders.head()

,order_id,customer_id,order_date,product_id,product_name,category,price,quantity,discount,payment_method,region,channel,record_status
0,ORD000001,C0291,2026-07-16,A004,토트백,패션,32000,5,0.00,카드,부산,Web,verified
1,ORD000002,C0254,2026-07-31,A003,캡,패션,25000,1,0.00,카드,서울,Web,verified
2,ORD000003,C0223,2026-07-05,S004,플래너,문구,12000,7,0.05,기타,서울,App,verified
3,ORD000004,C0074,2026-07-12,E002,마우스,전자기기,29000,3,0.05,카카오페이,경기,App,verified
4,ORD000005,C0049,2026-07-02,E005,USB 허브,전자기기,39000,5,0.05,네이버페이,경기,Web,verified


In [24]:
orders.shape

(1000, 13)

In [25]:
orders.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   order_id        1000 non-null   str    
 1   customer_id     1000 non-null   str    
 2   order_date      1000 non-null   str    
 3   product_id      1000 non-null   str    
 4   product_name    1000 non-null   str    
 5   category        1000 non-null   str    
 6   price           1000 non-null   int64  
 7   quantity        1000 non-null   int64  
 8   discount        960 non-null    float64
 9   payment_method  1000 non-null   str    
 10  region          985 non-null    str    
 11  channel         1000 non-null   str    
 12  record_status   1000 non-null   str    
dtypes: float64(1), int64(2), str(10)
memory usage: 170.7 KB


### Raw Data Evidence

- Raw 행 수: ______
- 열 수: ______
- 결측치가 있어 보이는 열: __________________
- 날짜처럼 보이는 열: __________________

## 어떤 열이 중요할까요?

팀장의 질문에 답하기 위해 먼저 필요하다고 생각하는 열을 적어보세요.

____________________________________________

In [ ]:
orders.columns

In [ ]:
# product_name, category, price, quantity 열만 선택하세요.
pass

# Mission 2 — 익숙한 조건부터

가격이 50,000원 이상인 주문을 찾습니다. 어떤 열을 어떤 값과 비교해야 하나요?

In [14]:
# 각 주문의 price가 50,000원 이상인지 True/False로 확인하세요.
pass
mask = orders['price'] >= 50000
len(orders[mask])

126

In [ ]:
# 가격이 50,000원 이상인 주문만 선택하세요.
pass

`score >= 80`과 `price >= 50000`은 어떤 점에서 같은가요?

In [15]:
# 전자기기 카테고리 주문만 선택하세요.
pass
df = orders = orders['category'] == '전자기기'
len(df)

1000

In [16]:
# 전자기기 주문 중 price가 100,000원 이상인 주문을 찾으세요.
pass
mask1 =(orders['category']=='전자기기') & (orders['price']>=100000)
orders[mask1]


KeyError: 'category'

서울에서 발생한 전자기기 주문을 찾으려면 어떤 두 조건이 필요할까요?

In [17]:
# 두 조건을 조합해 주문을 찾으세요.
pass
mask2 = (orders['region'] =='서울') & (orders['category']=='전자기기')
orders[mask2]

KeyError: 'region'

<details>
<summary>힌트 보기</summary>

region 조건과 category 조건을 각각 만든 뒤 `&`로 연결합니다.

</details>

# Mission 3 — 매출 열 만들기

현재 `sales` 열은 없습니다. 가격과 수량으로 할인 전 주문금액을 만들 수 있을까요?

In [33]:
# price × quantity를 계산하세요. 아직 새 열에 저장하지 마세요.
pass
orders["price"] * orders["quantity"]

0      160000
1       25000
2       84000
3       87000
4      195000
        ...  
995     70000
996     70000
997     52000
998    116000
999     36000
Length: 1000, dtype: int64

In [34]:
orders["sales"] = orders["price"] * orders["quantity"]
orders[["price", "quantity", "sales"]].head()

,price,quantity,sales
0,32000,5,160000
1,25000,1,25000
2,12000,7,84000
3,29000,3,87000
4,39000,5,195000


In [ ]:
# sales가 가장 큰 주문 5개를 확인하세요.
pass

## 할인까지 적용하면?

할인 적용 후 금액의 계산 구조는 `sales × (1 - discount)`입니다. 결과가 모두 숫자로 채워질지 먼저 예상하세요.

In [35]:
preview_final_sales = orders["sales"] * (1 - orders["discount"])
preview_final_sales.head(12)

0     160000.0
1      25000.0
2      79800.0
3      82650.0
4     185250.0
5      35100.0
6      49000.0
7      93100.0
8      49000.0
9      10800.0
10     78000.0
11     30000.0
dtype: float64

In [ ]:
preview_final_sales.isna().sum()

코드 문법은 맞지만 일부 계산 결과가 비어 있습니다. 원인을 데이터에서 찾아봅시다.

# Mission 4 — 이 데이터를 믿어도 될까?

In [49]:
# 각 열의 결측치 개수를 확인하세요.
pass
orders.isna()

,order_id,customer_id,order_date,product_id,product_name,category,price,quantity,discount,payment_method,region,channel,record_status,sales,final_sales
0,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
4,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
995,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
996,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
997,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
998,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False


<details>
<summary>힌트 보기</summary>

`isna()`의 결과를 열별로 더해보세요.

</details>

In [50]:
orders.isna().sum()

order_id           0
customer_id        0
order_date         0
product_id         0
product_name       0
category           0
price              0
quantity           0
discount          40
payment_method     0
region            15
channel            0
record_status      0
sales              0
final_sales       40
dtype: int64

### Missing Evidence

- discount 결측: ______
- region 결측: ______

## 같은 NaN, 다른 의미

`discount`가 비어 있으면 할인 적용 기록이 없는 주문입니다. 반면 `region`이 비어 있으면 지역 정보가 기록되지 않은 상태입니다.

두 값을 똑같이 처리해야 할까요? 각각의 처리와 이유를 적어보세요.

### discount

처리: ____모든 값을 0처리________________  
이유: ____________________

### region

처리: ____________________  
이유: ____________________

In [ ]:
# discount의 결측치를 업무 의미에 맞게 처리하세요.
pass

In [53]:
orders["discount"] = orders["discount"].fillna(0) #discount에 na를 0으로 채우기
orders["discount"].isna().sum() #discount의 na값 갯수 다시 확인

np.int64(0)

## S47 — final_sales 만들기

이 단계에서는 각 행의 계산이 더 이상 결측 때문에 비지 않는지만 확인합니다. 아직 전체 공식 합계를 확정하지 않습니다.

In [51]:
# sales와 discount로 final_sales 열을 만드세요.
pass
print(orders.columns)

Index(['order_id', 'customer_id', 'order_date', 'product_id', 'product_name',
       'category', 'price', 'quantity', 'discount', 'payment_method', 'region',
       'channel', 'record_status', 'sales', 'final_sales'],
      dtype='str')


In [ ]:
## final_sales 업데이트
orders["final_sales"] = orders["sales"] * (1 - orders["discount"]) #다시 계산해서 업데이트
orders["final_sales"].isna().sum() #final_sales의 na값 확인

np.int64(0)

## 날짜처럼 보이는 값은 정말 날짜일까요?

화면 모양과 실제 dtype은 다를 수 있습니다.

In [60]:
# order_date의 dtype을 확인하세요.
pass
print(type(orders.order_date))
print(orders['order_date'].dtype) #fd 컬럼 시리즈의 타입확인

pd.to_datetime(orders["order_date"]) #데이터 타입 변환

<class 'pandas.Series'>
str


0     2026-07-16
1     2026-07-31
2     2026-07-05
3     2026-07-12
4     2026-07-02
         ...    
995   2026-07-14
996   2026-07-30
997   2026-07-22
998   2026-07-03
999   2026-07-06
Name: order_date, Length: 1000, dtype: datetime64[us]

In [61]:
orders["order_date"] = pd.to_datetime(orders["order_date"])
orders["order_date"].dtype

dtype('<M8[us]')

## 지역 값 살펴보기

지역별 개수를 세어 같은 지역이 서로 다른 값처럼 저장된 경우가 있는지 찾아보세요.

In [62]:
# NaN을 포함해 region별 값의 개수를 확인하세요.
pass
orders['region'].isna().sum()

np.int64(15)

In [67]:
print(orders["region"].value_counts()) #기본값 na 제외
#==> print(orders["region"].value_counts(dropna=True))
orders["region"].value_counts(dropna=False) #na값을 포함한 갯수

region
서울      339
경기      216
부산      148
대전      108
대구       78
광주       71
서울       10
 경기       5
부산        4
 대전       3
대구        2
 광주       1
Name: count, dtype: int64


region
서울      339
경기      216
부산      148
대전      108
대구       78
광주       71
NaN      15
서울       10
 경기       5
부산        4
 대전       3
대구        2
 광주       1
Name: count, dtype: int64

사람 눈에는 같은 지역인데 Pandas가 다르게 센 값이 있나요? 왜 그럴까요?

In [69]:
# region의 앞뒤 공백을 제거하세요.
pass
orders['region'].str.strip()

0      부산
1      서울
2      서울
3      경기
4      경기
       ..
995    부산
996    서울
997    서울
998    대전
999    부산
Name: region, Length: 1000, dtype: str

<details>
<summary>힌트 보기</summary>

문자열의 앞뒤 공백은 `.str.strip()`으로 정리할 수 있습니다.

</details>

In [70]:
orders["region"] = orders["region"].str.strip()
dirty_variants = ["서울 ", " 경기 ", "부산 ", " 대전", "대구  ", " 광주"]
orders["region"].isin(dirty_variants).sum()

np.int64(0)

공백이 포함된 dirty variant가 0개가 되었는지만 확인했습니다. 지역별 공식 개수는 아직 확정하지 않습니다.

## 같은 주문이 두 번 저장되었다면?

먼저 모든 값이 완전히 같은 행이 있는지 확인합니다.

In [103]:
# 완전히 같은 행이 몇 개인지 확인하세요.
pass
print(orders.duplicated()  ) #==> True / False 반환
print("전체행 중복 갯수:" ,orders.duplicated().sum())
#subset : 특정 열만을 대상으로 할 수 있습니다. list의 사용도 가능합니다.
#keep = True -->> 기본 중복결과 제거후 출력
#keep = False -->> 중복결과 모두출력
orders[orders.duplicated(keep=False)].sort_values('order_id')


0      False
1      False
2      False
3      False
4      False
       ...  
995     True
996     True
997    False
998    False
999    False
Length: 1000, dtype: bool
전체행 중복 갯수: 5


,order_id,customer_id,order_date,product_id,product_name,category,price,quantity,discount,payment_method,region,channel,record_status,sales,final_sales
127,ORD000128,C0213,2026-07-11,E004,웹캠,전자기기,89000,5,0.20,카카오페이,서울,Web,verified,445000,356000.0
992,ORD000128,C0213,2026-07-11,E004,웹캠,전자기기,89000,5,0.20,카카오페이,서울,Web,verified,445000,356000.0
256,ORD000257,C0140,2026-07-08,S001,노트,문구,4500,1,0.05,카드,경기,Web,verified,4500,4275.0
993,ORD000257,C0140,2026-07-08,S001,노트,문구,4500,1,0.05,카드,경기,Web,verified,4500,4275.0
440,ORD000441,C0260,2026-07-27,L006,머그컵,생활,14000,3,0.00,네이버페이,서울,Web,verified,42000,42000.0
994,ORD000441,C0260,2026-07-27,L006,머그컵,생활,14000,3,0.00,네이버페이,서울,Web,verified,42000,42000.0
687,ORD000688,C0091,2026-07-14,L005,쿠션,생활,35000,2,0.00,카드,부산,Web,verified,70000,70000.0
995,ORD000688,C0091,2026-07-14,L005,쿠션,생활,35000,2,0.00,카드,부산,Web,verified,70000,70000.0
902,ORD000903,C0288,2026-07-30,L005,쿠션,생활,35000,2,0.00,카드,서울,App,verified,70000,70000.0
996,ORD000903,C0288,2026-07-30,L005,쿠션,생활,35000,2,0.00,카드,서울,App,verified,70000,70000.0


In [71]:
orders.duplicated().sum()

np.int64(5)

In [104]:
# 중복된 행들을 직접 확인하세요.
pass
print("order_id 중복 갯수:", orders.duplicated().sum())
print(orders[orders.duplicated()]) #--> 중복된 값을 유니크하게 출력

orders[orders['order_id'].str.strip() =='ORD000128'] #--> 예> 중복된 키값으로 조회시 2개 중복으로 확인


order_id 중복 갯수: 5
      order_id customer_id order_date product_id product_name category  price  \
992  ORD000128       C0213 2026-07-11       E004           웹캠     전자기기  89000   
993  ORD000257       C0140 2026-07-08       S001           노트       문구   4500   
994  ORD000441       C0260 2026-07-27       L006          머그컵       생활  14000   
995  ORD000688       C0091 2026-07-14       L005           쿠션       생활  35000   
996  ORD000903       C0288 2026-07-30       L005           쿠션       생활  35000   

     quantity  discount payment_method region channel record_status   sales  \
992         5      0.20          카카오페이     서울     Web      verified  445000   
993         1      0.05             카드     경기     Web      verified    4500   
994         3      0.00          네이버페이     서울     Web      verified   42000   
995         2      0.00             카드     부산     Web      verified   70000   
996         2      0.00             카드     서울     App      verified   70000   

     final_sales  
9

,order_id,customer_id,order_date,product_id,product_name,category,price,quantity,discount,payment_method,region,channel,record_status,sales,final_sales
127,ORD000128,C0213,2026-07-11,E004,웹캠,전자기기,89000,5,0.2,카카오페이,서울,Web,verified,445000,356000.0
992,ORD000128,C0213,2026-07-11,E004,웹캠,전자기기,89000,5,0.2,카카오페이,서울,Web,verified,445000,356000.0


## Row Duplicate와 Key Duplicate

- Row Duplicate: 모든 값이 같은 행이 반복됨
- Key Duplicate: `order_id`는 같지만 다른 값이 서로 다를 수 있음

In [105]:
# order_id가 중복된 주문이 있는지 확인하세요.
pass
print("중복주문 ID:",orders['order_id'].duplicated().sum())

orders[orders['order_id'].duplicated()].sort_values('order_id') #keep 옵션값 기본 True --> 유니크게 결과 출력
orders[orders['order_id'].duplicated(keep=False)].sort_values('order_id') #keep = False 모든 값 출력

중복주문 ID: 8


,order_id,customer_id,order_date,product_id,product_name,category,price,quantity,discount,payment_method,region,channel,record_status,sales,final_sales
127,ORD000128,C0213,2026-07-11,E004,웹캠,전자기기,89000,5,0.20,카카오페이,서울,Web,verified,445000,356000.0
992,ORD000128,C0213,2026-07-11,E004,웹캠,전자기기,89000,5,0.20,카카오페이,서울,Web,verified,445000,356000.0
172,ORD000173,C0160,2026-07-22,F006,견과류,식품,13000,3,0.10,카드,서울,Web,verified,39000,35100.0
997,ORD000173,C0160,2026-07-22,F006,견과류,식품,13000,4,0.10,기타,서울,App,ingestion_error,52000,46800.0
256,ORD000257,C0140,2026-07-08,S001,노트,문구,4500,1,0.05,카드,경기,Web,verified,4500,4275.0
993,ORD000257,C0140,2026-07-08,S001,노트,문구,4500,1,0.05,카드,경기,Web,verified,4500,4275.0
440,ORD000441,C0260,2026-07-27,L006,머그컵,생활,14000,3,0.00,네이버페이,서울,Web,verified,42000,42000.0
994,ORD000441,C0260,2026-07-27,L006,머그컵,생활,14000,3,0.00,네이버페이,서울,Web,verified,42000,42000.0
513,ORD000514,C0088,2026-07-03,E002,마우스,전자기기,29000,3,0.00,카카오페이,대전,App,verified,87000,87000.0
998,ORD000514,C0088,2026-07-03,E002,마우스,전자기기,29000,4,0.00,기타,대전,Web,ingestion_error,116000,116000.0


In [ ]:
print(orders["order_id"].duplicated(keep=False))
#.loc는 bool 배열과 함께 사용이 가능합니다. --> True / False 에 해당하는 데이터 출력
print(orders.loc[orders["order_id"].duplicated()].sort_values("order_id")) #keep = True 중복제거
orders.loc[orders["order_id"].duplicated(keep=False)].sort_values("order_id")  #keep = False 모두

0      False
1      False
2      False
3      False
4      False
       ...  
995     True
996     True
997     True
998     True
999     True
Name: order_id, Length: 1000, dtype: bool
      order_id customer_id order_date product_id product_name category  price  \
992  ORD000128       C0213 2026-07-11       E004           웹캠     전자기기  89000   
997  ORD000173       C0160 2026-07-22       F006          견과류       식품  13000   
993  ORD000257       C0140 2026-07-08       S001           노트       문구   4500   
994  ORD000441       C0260 2026-07-27       L006          머그컵       생활  14000   
998  ORD000514       C0088 2026-07-03       E002          마우스     전자기기  29000   
995  ORD000688       C0091 2026-07-14       L005           쿠션       생활  35000   
999  ORD000821       C0108 2026-07-06       F001           원두       식품  18000   
996  ORD000903       C0288 2026-07-30       L005           쿠션       생활  35000   

     quantity  discount payment_method region channel    record_status  \
992        

,order_id,customer_id,order_date,product_id,product_name,category,price,quantity,discount,payment_method,region,channel,record_status,sales,final_sales
127,ORD000128,C0213,2026-07-11,E004,웹캠,전자기기,89000,5,0.20,카카오페이,서울,Web,verified,445000,356000.0
992,ORD000128,C0213,2026-07-11,E004,웹캠,전자기기,89000,5,0.20,카카오페이,서울,Web,verified,445000,356000.0
172,ORD000173,C0160,2026-07-22,F006,견과류,식품,13000,3,0.10,카드,서울,Web,verified,39000,35100.0
997,ORD000173,C0160,2026-07-22,F006,견과류,식품,13000,4,0.10,기타,서울,App,ingestion_error,52000,46800.0
256,ORD000257,C0140,2026-07-08,S001,노트,문구,4500,1,0.05,카드,경기,Web,verified,4500,4275.0
993,ORD000257,C0140,2026-07-08,S001,노트,문구,4500,1,0.05,카드,경기,Web,verified,4500,4275.0
440,ORD000441,C0260,2026-07-27,L006,머그컵,생활,14000,3,0.00,네이버페이,서울,Web,verified,42000,42000.0
994,ORD000441,C0260,2026-07-27,L006,머그컵,생활,14000,3,0.00,네이버페이,서울,Web,verified,42000,42000.0
513,ORD000514,C0088,2026-07-03,E002,마우스,전자기기,29000,3,0.00,카카오페이,대전,App,verified,87000,87000.0
998,ORD000514,C0088,2026-07-03,E002,마우스,전자기기,29000,4,0.00,기타,대전,Web,ingestion_error,116000,116000.0


충돌 레코드의 `record_status`를 확인하세요. 단순히 먼저 나온 행을 고르는 대신, 어떤 레코드를 신뢰할지 근거를 사용해야 합니다.

# Analysis-Ready Gate

이제 raw 1,000행에서 분석에 사용할 논리 주문으로 상태를 전환합니다.

`orders`는 원본에서 시작한 작업 데이터이고, `logical_orders`는 중복과 충돌 판단을 반영한 분석용 데이터입니다.

In [ ]:
verified_orders = orders.loc[
    orders["record_status"].ne("ingestion_error")
].copy()

logical_orders = (
    verified_orders
    .drop_duplicates(subset="order_id", keep="first")
    .copy()
)

pd.Series({
    "logical_rows": len(logical_orders),
    "unique_order_id": logical_orders["order_id"].nunique(),
})

행 수와 고유 주문 ID 수가 모두 992인지 확인한 뒤에만 공식 분석을 시작합니다.

In [ ]:
gross_total = logical_orders["sales"].sum()
discount_total = (logical_orders["sales"] - logical_orders["final_sales"]).sum()
final_total = logical_orders["final_sales"].sum()

pd.Series({
    "gross_sales": int(gross_total),
    "discount_amount": int(discount_total),
    "final_sales": int(final_total),
})

### 첫 공식 매출 Evidence

- Gross Sales: **78,000,000원**
- Discount Amount: **4,000,000원**
- Final Sales: **74,000,000원**

이 숫자들은 `logical_orders`가 992개의 고유 주문으로 확정된 뒤 처음 사용합니다.

# Cleaning Evidence

| 문제 | 발견 방법 | 처리 | 이유 |
|---|---|---|---|
| discount 결측 | | | |
| region 결측 | | | |
| region 공백 | | | |
| 완전 중복 | | | |
| order_id 충돌 | | | |
| order_date dtype | | | |

# Mission 5 — 무엇이 매출을 만들었을까?

전체 매출 하나만으로는 어떤 카테고리가 매출을 만들었는지 알 수 없습니다. 카테고리별로 나누어 비교합니다.

In [ ]:
logical_orders["final_sales"].sum()

카테고리가 100개라면 각 카테고리를 따로 필터링해 합계를 100번 계산하는 방법은 불편합니다.

In [ ]:
logical_orders.groupby("category")

`데이터 → category별로 나눔 → 각 그룹에서 계산 → 결과 비교`

카테고리별 합계 결과가 몇 개 나올지 예상해보세요.

In [40]:
logical_orders.groupby("category")["final_sales"].sum()

NameError: name 'logical_orders' is not defined

`category`별로 묶고, `final_sales`를 선택하고, 각 그룹의 합을 계산했습니다.

In [ ]:
# 카테고리별 매출을 높은 순서대로 정렬하세요.
pass

## '가장 많이 팔린 카테고리'는 무슨 뜻일까요?

1. 주문 건수가 가장 많은 카테고리
2. 총매출이 가장 큰 카테고리

두 답이 같은지 확인하세요.

In [ ]:
# logical_orders에서 카테고리별 주문 건수를 구하세요.
pass

In [ ]:
# logical_orders에서 카테고리별 총매출을 구하세요.
pass

In [39]:
category_summary = (
    logical_orders
        .groupby("category")["final_sales"]
        .agg(["count", "sum", "mean"])
)
category_summary

NameError: name 'logical_orders' is not defined

### Category Evidence

주문 건수 1위: __________________  
총매출 1위: __________________  
평균 주문금액 1위: __________________

`count`, `sum`, `mean`은 각각 어떤 질문에 답하나요?

In [38]:
# 평균 주문금액이 가장 높은 카테고리를 찾으세요.
pass

print(orders.columns)
channel_summary =(
            orders
                .groupby('channel')['price'*'quantity']
                .agg('count','sum','mean')
)
channel_summary


Index(['order_id', 'customer_id', 'order_date', 'product_id', 'product_name',
       'category', 'price', 'quantity', 'discount', 'payment_method', 'region',
       'channel', 'record_status', 'sales'],
      dtype='str')


TypeError: can't multiply sequence by non-int of type 'str'

# Applied Challenge — App과 Web 중 어떤 채널이 더 중요할까요?

바로 답하지 말고, 먼저 '중요하다'를 어떤 숫자로 판단할지 정하세요.

1. ____________________
2. ____________________
3. ____________________

In [ ]:
# logical_orders를 기준으로 채널별 주문 건수, 총매출, 평균 주문금액을 비교하세요.
pass

<details>
<summary>힌트 보기</summary>

카테고리에서 사용한 `groupby + agg` 구조를 channel에 다시 적용할 수 있습니다.

</details>

## Channel Evidence

### 주문 건수가 많은 채널
____________________ / 근거: ____________________

### 총매출이 높은 채널
____________________ / 근거: ____________________

### 평균 주문금액이 높은 채널
____________________ / 근거: ____________________

### 내가 더 중요하다고 본 채널과 이유
________________________________________________

# 이번 Notebook에서 한 일

`업무 질문 → 데이터 확인 → 이상 발견 → 의미 판단 → 정리 → Analysis-Ready Gate → 그룹 비교 → Evidence`

다음에는 주문 파일에 없는 고객 등급을 다른 파일에서 연결합니다.